### ISSUE TITLE
About over fitting

### ISSUE THREAD

**Issue by eeyrw** [cfzd/Ultra-Fast-Lane-Detection#5]
I notice that in the final stage of the network all feature maps are reorganized for FC layer. I think such kind of operation may cause the spatial infomation loss, which will result to overfitting on distrbution of certain dataset. Does it actually happen?

**Comment by cfzd** [comment]
@eeyrw 
Hi, thanks for your question. In fact, there are four problems.

+ would FC operation lose the spatial information loss? 

    I don't think so since many two-stage detection methods like Fast RCNN also uses  FC layers to regress the bounding box of objects, which requires precise spatial information. These methods work well and have been well recognized.

+ would  FC layers result in overfitting?

    I don't think there are any connections between FC layers and overfitting. If an FC layer overfits, why not a convolutional layer overfits? In many works,  1x1 Conv is considered equivalent to FC operation.

+ whether there is an overfitting problem?  
    
    The overfitting problem indeed exists, but it is caused by the dataset itself, not our method. The patterns of lanes are very likely to be similar as we all know, and this is the reason for overfitting.

+ how do we deal with the overfitting problem?

    To deal with the overfitting and make the patterns of lanes more varied, we use the data augmentation proposed in section 4.1, data augmentation part in our paper. With integrated random rotation, random horizontal and vertical translation, the patterns of lanes are much more diverse. The corresponding code can be found [here](https://github.com/cfzd/Ultra-Fast-Lane-Detection/blob/b4f8d84cb464b11ef0cbc3e9ca45ad1594660616/data/dataloader.py#L23).

**Comment by eeyrw** [comment]
@cfzd Thank for your informative reply. Your points are so solid and there is nothing to worry with regard to the overfitting caused by the network itself.
